packages/maskbook/src/plugins/Trader/apis/coinmarketcap/getPriceStat.ts

```diff
@@ -0,0 +1,53 @@
+import { CMC_LATEST_BASE_URL } from '../../constants/trending'
+
+export interface PriceState {
+    id: number
+    mc: number
+    p: number
+    p1h: number
+    p7d: number
+    p24h: number
+    v: number
+}
+
+function openWebSocket() {
+    return new Promise<WebSocket>((resolve, reject) => {
+        try {
+            const client = new WebSocket(CMC_LATEST_BASE_URL)
+            return resolve(client)
+        } catch (error) {
+            return reject(error)
+        }
+    })
+}
+
+export function getPriceStat(coinId: string) {
+    return new Promise<PriceState>((resolve, reject) => {
+        openWebSocket()
+            .then((client) => {
+                client.onopen = () => {
+                    const params = {
+                        method: 'subscribe',
+                        id: 'price',
+                        data: {
+                            cryptoIds: [coinId],
+                        },
+                    }
+                    client.send(JSON.stringify(params))
+                }
+                client.onmessage = (ev) => {
+                    if (typeof ev.data !== 'string') {
+                        return
+                    }
+                    try {
+                        const prices = JSON.parse(ev.data)
+                        resolve(prices['d']['cr'])
```

Comment:
@Jack-Works async-rpc framework support non-standard json rpc field definition ?

Reply:
Nope, what does these fields stand for?

Reply:
this data is websocket's message, not async-rpc